api_tests/lib_sdk/wallets/lightning.ts

```diff
@@ -0,0 +1,102 @@
+import { pollUntilMinted, Wallet } from "./index";
+import { Asset } from "../asset";
+import BigNumber from "bignumber.js";
+import BitcoinRpcClient from "bitcoin-core";
+import { BitcoinNodeConfig } from "../../lib/bitcoin";
+import { toBitcoin } from "satoshi-bitcoin";
+import { Lnd } from "../lnd";
+import { HarnessGlobal } from "../../lib/util";
+import { Logger } from "log4js";
+import { E2ETestActorConfig } from "../../lib/config";
+
+declare var global: HarnessGlobal;
+
+export class LightningWallet implements Wallet {
+    public static async newInstance(
+        bitcoinNodeConfig: BitcoinNodeConfig,
+        logger: Logger,
+        logDir: string,
+        actorConfig: E2ETestActorConfig
+    ) {
+        const bitcoinRpcClient = new BitcoinRpcClient({
+            network: bitcoinNodeConfig.network,
+            port: bitcoinNodeConfig.rpcPort,
+            host: bitcoinNodeConfig.host,
+            username: bitcoinNodeConfig.username,
+            password: bitcoinNodeConfig.password,
+        });
+
+        const bitcoindDataDir = global.bitcoind.getDataDir();
+
+        const lnd = new Lnd(logger, logDir, actorConfig, bitcoindDataDir);
+        await lnd.start();
+
+        return new LightningWallet(lnd, bitcoinRpcClient);
+    }
+
+    public MaximumFee = 0;
+
+    private constructor(
+        public readonly inner: Lnd,
+        private readonly bitcoinRpcClient: BitcoinRpcClient
+    ) {}
+
+    public async mint(asset: Asset): Promise<void> {
+        if (asset.name !== "bitcoin") {
+            throw new Error(
+                `Cannot mint asset ${asset.name} with BitcoinWallet`
+            );
+        }
+
+        const startingBalance = new BigNumber(
+            await this.getBalanceByAsset(asset)
+        );
+
+        const minimumExpectedBalance = new BigNumber(asset.quantity);
+
+        const blockHeight = await this.bitcoinRpcClient.getBlockCount();
+        if (blockHeight < 101) {
+            throw new Error(
+                "unable to mint bitcoin, coinbase transactions are not yet spendable"
+            );
+        }
+
+        await this.bitcoinRpcClient.sendToAddress(
+            await this.address(),
+            toBitcoin(minimumExpectedBalance.times(2).toString()) // make sure we have at least twice as much
+        );
+
+        await pollUntilMinted(
+            this,
+            startingBalance.plus(minimumExpectedBalance),
+            asset
+        );
+    }
+
+    public async address(): Promise<string> {
+        return this.inner.createChainAddress();
+    }
+
+    public async getBalanceByAsset(asset: Asset): Promise<BigNumber> {
```

Comment:
I think this should only return the channel balance.

Reply:
Let's fix that once I assert paid invoice.